experimental/trusted_shuffler/trusted_shuffler/src/tests.rs

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::TrustedShuffler;
+use futures::future::join_all;
+use std::{sync::Arc, time::Duration};
+use tokio::time::timeout;
+
+const TEST_ANONYMITY_VALUE: usize = 10;
+const TEST_TIMEOUT_MILLISECONDS: u64 = 100;
+
+// A test function that processes requests.
+// In the real use-case, this function should send requests to the backend and
+// return responses.
+async fn handle_request(request: Vec<u8>) -> Vec<u8> {
+    generate_response(request)
+}
+
+// Non-async version of the `handle_request` function used to create expected responses.
+fn generate_response(request: Vec<u8>) -> Vec<u8> {
+    let parsed_request = String::from_utf8(request).expect("Couldn't parse request");
+    format!("Response for: {}", parsed_request)
+        .to_string()
+        .as_bytes()
+        .to_vec()
+}
+
+// Generates a request and a corresponding response from a string.
+fn generate_request_response(data: &str) -> (Vec<u8>, Vec<u8>) {
```

Comment:
Here the respone only echoes the request, no? Is there any reason to have that for the future? Or am I missing something?

Reply:
Ah, I see now! Thanks! 

Does it make sense to call it `generate_test_request_response`? 

And maybe rename `response` to `expected_response`. We are creating the request/expected_response pair here, no?

Reply:
+1 it took me a while to understand why a request is being generated twice.